packages/typeorm-decorator-ismoney/src/index.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  registerDecorator,
+  ValidationArguments,
+  ValidationOptions,
+  ValidatorConstraint,
+  ValidatorConstraintInterface,
+} from 'class-validator';
+
+import codes from './codes';
+import { IMoney } from './Money';
+
+const settingsDefault = {
+  allowNegative: false,
+  min: Number.MIN_SAFE_INTEGER,
+  max: Number.MAX_SAFE_INTEGER,
+  currencies: null,
+};
+
+interface IMoneyValidationOptions {
+  allowNegative?: boolean;
+  min?: number;
+  max?: number;
+  currencies?: string[];
+}
+
+@ValidatorConstraint({ name: 'IsMoney', async: false })
+export class IsMoneyValidator implements ValidatorConstraintInterface {
+  public validate(money: IMoney, args: ValidationArguments) {
+    const { amount, currency } = money;
+    const settingsOrDefault = args.constraints[0] || settingsDefault;
```

Comment:
why is it needed if then use Object.assign with the same values?

Reply:
can you use` Object.assign({}, settingsDefault, args.constraints[0])` ?

Reply:
Actually this can be shortened. That's the remains of destructuring.

Reply:
Fixed. Also, made the options optional.